lib/shopping_cart.rb

```diff
@@ -0,0 +1,61 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity
+    @products = []
+    @details = Hash.new
+  end
+
+  def name
+    @name
+  end
+
+  def capacity
+    @capacity[0..1].to_i
+  end
```

Comment:
if you're going to have a method for capacity (which I like since you're formatting what was passed in), I'd remove the attr_reader for capacity since that's creating a method to read the attribute that was passed in.